language/stdlib/modules/AccountLimits.move

```diff
@@ -165,26 +157,41 @@ module AccountLimits {
         new_max_inflow: u64,
         new_max_outflow: u64,
         new_max_holding_balance: u64,
+        new_time_period: u64,
     ) acquires LimitsDefinition {
-        assert(has_treasury_compliance_role(tc_account), ENOT_TREASURY_COMPLIANCE);
+        assert(Roles::has_treasury_compliance_role(tc_account), ENOT_TREASURY_COMPLIANCE);
         // As we don't have Optionals for txn scripts, in update_account_limit_definition.move
         // we use 0 value to represent a None (ie no update to that variable)
         let limits_def = borrow_global_mut<LimitsDefinition<CoinType>>(limit_address);
         if (new_max_inflow > 0) { limits_def.max_inflow = new_max_inflow };
         if (new_max_outflow > 0) { limits_def.max_outflow = new_max_outflow };
         if (new_max_holding_balance > 0) { limits_def.max_holding = new_max_holding_balance };
+        if (new_time_period > 0) { limits_def.time_period = new_time_period };
     }
 
-    /// Since we don't track balances of accounts before they are limited, once
-    /// they do become limited the approximate balance in `CointType` held by
-    /// the entity across all of its accounts will need to be set by the association.
-    public fun set_current_holdings<CoinType>(
+    /// Update either the `tracked_balance` or `limit_address` fields of the
+    /// `Window<CoinType>` stored under `window_address`.
+    /// * Since we don't track balances of accounts before they are limited, once
```

Comment:
This is a really nasty problem, I didn't fully understand it in the diff that encountered the performance problem while trying to do this the right way. No suggestions for improvement at the moment, but thinking...

Reply:
It really is quite nasty. The only way I can think is to have the windows be held in child VASP accounts, but then that doesn't solve the issue of collecting the balances for all the child accounts...I think it's definitely something worth brainstorming about at some point.